### PR TITLE
Bringing in full airbnb rules for react

### PR DESCRIPTION
**Pull request by InsaneViku**
As per discussion, I'm bringing in the airbnb react rulesets. The benefits outweigh the cons IMO. We can disable more rules as we see fit, but I feel the collection of rules ootb will help us immediately.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -34,11 +35,11 @@ module.exports = {
     // Allow to use `_` in functions and variable names
     'no-underscore-dangle': OFF,
 
-    // Set the max length of a line to 200 characters
+    // Set the max length of a line to 120 characters
```
Comment (InsaneViku) [inline index.js]: I feel 200 is too long - I've been accustomed to conventions of 100-120 limits. Airbnb itself has a 100 limit, so I think 120 will be fair for us.

File: package.json [hunk 2]
```diff
@@ -23,8 +23,11 @@
     "eslint-plugin-import": "^2.16.0"
   },
   "optionalDependencies": {
+    "eslint-config-airbnb": "^17.1.0",
     "eslint-config-react-app": "^3.0.8",
```
Comment (minhtule) [inline package.json]: As discussed, I think we should have only 1 guideline so it's easier for new members to read up and follow.

Comment (InsaneViku) [reply package.json]: done - gutted react-app

File: package.json [hunk 3]
```diff
@@ -23,15 +23,19 @@
...
   },
   "dependencies": {
     "eslint-config-airbnb-base": "^13.1.0"
   },
   "devDependencies": {
     "babel-eslint": "^9.0.0",
     "eslint": "^5.15.1",
-    "jest": "^24.5.0"
+    "jest": "^24.5.0",
+    "jest-haste-map": "^24.5.0",
```
Comment (InsaneViku) [inline package.json]: these were missing, so popped them in

**Pull request merged**